src/data/events/EventsD2ApiRepository.ts

```diff
@@ -237,53 +243,50 @@ export class EventsD2ApiRepository implements EventsRepository {
         }
     }
 
-    private cleanEventsImportResponse(importResult: EventsPostResponse): SynchronizationResult {
-        const errors = _(importResult.importSummaries)
-            .flatMap(element => {
-                if (element.status !== "ERROR") return undefined;
-                return (
-                    element.conflicts?.map(({ object, value }) => ({
-                        id: element.reference ?? "",
-                        message: _([element.description, object, value]).compact().join(" "),
-                    })) ?? [{ id: element.reference ?? "", message: element.description }]
-                );
-            })
-            .compact()
-            .value();
-
-        const stats: SynchronizationStats = _.pick(importResult, [
-            "imported",
-            "updated",
-            "ignored",
-            "deleted",
-            "total",
-        ]);
-
+    private cleanEventsImportResponse(importResult: TrackerPostResponse): SynchronizationResult {
         return {
-            status: importResult.status,
-            stats,
+            status: importResult.status === "OK" ? "SUCCESS" : importResult.status,
+            stats: {
+                imported: importResult.stats.created,
+                updated: importResult.stats.updated,
+                ignored: importResult.stats.ignored,
+                deleted: importResult.stats.deleted,
+                total: importResult.stats.total,
+            },
             instance: this.instance.toPublicObject(),
-            errors,
+            errors: importResult.validationReport.errorReports.map(error => {
+                return {
+                    id: error.uid,
+                    message: error.message,
+                    type: error.trackerType,
+                };
+            }),
             date: new Date(),
             type: "events",
             response: importResult,
         };
     }
 
-    private async push(params: DataImportParams, data: EventsPackage): Promise<SynchronizationResult> {
-        const eventsPostParams: EventsPostParams = {
+    private async push(params: DataImportParams, events: ProgramEvent[]): Promise<SynchronizationResult> {
+        const eventsPostParams: TrackerPostParams = {
             idScheme: params.idScheme ?? "UID",
             dataElementIdScheme: params.dataElementIdScheme ?? "UID",
             orgUnitIdScheme: params.orgUnitIdScheme ?? "UID",
-            dryRun: params.dryRun ?? false,
-            preheatCache: params.preheatCache ?? false,
-            skipExistingCheck: params.skipExistingCheck ?? false,
+            importMode: params.importMode ?? "COMMIT",
+
+            // TODO: Check if these params exists in the new tracker endpoint
+            // preheatCache: params.preheatCache ?? false,
+            // skipExistingCheck: params.skipExistingCheck ?? false,
```

Comment:
I don't see any references to these two:

https://github.com/dhis2/dhis2-core/blob/2.40.4/dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/imports/RequestParams.java#L50